Document the preference of using HSMs than other forms of securing private keys

aronowski opened the pull request:
As per the call on October 7, 2024, documenting why HSMs should be used
instead of other methods of protecting private keys.

Showcasing, how important HSMs are, might be realized as a
community-backed venue for helping out setting these, especially when
workarounds are needed for some of the problematic tokens. Add an
example entry meant to be run in a test environment and link to it in
the docs.

aronowski committed:
WIP: a bit about why HSMs are important

Signed-off-by: Kamil Aronowski <kamil.aronowski@yahoo.com>

aronowski committed:
Add proposal for HSM Hardware Compatibility List venue

Showcasing, how important HSMs are, might be realized as a
community-backed venue for helping out setting these, especially when
workarounds are needed for some of the problematic tokens. Add an
example entry meant to be run in a test environment and link to it in
the docs.

Signed-off-by: Kamil Aronowski <kamil.aronowski@yahoo.com>

steve-mcintyre commented on docs/submitting.md:
```diff
@@ -109,9 +109,20 @@ There are strong recommendations on the types of certificates that
 security. See the 12th bullet in [Microsoft UEFI Signing
 Requirements](https://techcommunity.microsoft.com/t5/hardware-dev-center/updated-uefi-signing-requirements/ba-p/1062916)
-for tips on this, and also some sugested schems for key and certificate
+for tips on this, and also some suggested schems for key and certificate
```
"schemes" maybe?

steve-mcintyre commented:
I'm not sure the softhsm example setup belongs here - maybe better as a personal blog / repo?

aronowski committed:
Fix one typo in docs/submitting.md

"schems" -> "schemes", as per PR #448

Signed-off-by: Kamil Aronowski <kamil.aronowski@yahoo.com>

aronowski committed:
Remove the HCL proposal

As discussed, this might work better as an unofficial post, than
official venue.

Signed-off-by: Kamil Aronowski <kamil.aronowski@yahoo.com>
